docs/topics/testing/advanced.txt

```diff
@@ -600,9 +600,13 @@ utility methods in the ``django.test.utils`` module.
 
 .. function:: setup_test_environment()
 
-    Performs any global pre-test setup, such as the installing the
-    instrumentation of the template rendering system and setting up
-    the dummy email outbox.
+    Performs global pre-test setup, like installing instrumentation for the
+    template rendering system and setting up the dummy email outbox.
+
+    This function also saves the values of settings that it changes so that
```

Comment:
I updated this patch and also created a new PR for raising the error here: https://github.com/django/django/pull/7032

Let me know if I need to create a ticket.